File: sdk/kotlin/src/main/kotlin/com/telecom/SystemAPI.kt
package com.telecom

/**
 * API for system management
 */
class SystemAPI(private val client: HTTPClient) {
    
    /**
     * Get system statistics
     */
    suspend fun getStats(): SystemStats {
        return client.get("/v1/system/stats")
    }
    
    /**
     * Get system health status
     */
    suspend fun getHealth(): HealthStatus {
        return client.get("/v1/health")
    }
}

File: sdk/go/system_api.go
package telecom

import (
	"context"
)

// SystemAPI handles system-related API calls
type SystemAPI struct {
	client *HTTPClient
}

// NewSystemAPI creates a new SystemAPI
func NewSystemAPI(client *HTTPClient) *SystemAPI {
	return &SystemAPI{client: client}
}

// GetStats retrieves system statistics
func (s *SystemAPI) GetStats(ctx context.Context) (*SystemStats, error) {
	var stats SystemStats
	err := s.client.Get(ctx, "/v1/system/stats", &stats)
	return &stats, err
}

// GetHealth retrieves system health status
func (s *SystemAPI) GetHealth(ctx context.Context) (*HealthStatus, error) {
	var health HealthStatus
	err := s.client.Get(ctx, "/v1/health", &health)
	return &health, err
}
